core/src/test/java/com/linecorp/armeria/server/RedirectServiceTest.java

```diff
@@ -0,0 +1,368 @@
+/*
+ * Copyright 2018 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.net.ServerSocket;
+
+import org.junit.Rule;
+import org.junit.Test;
+
+import com.linecorp.armeria.client.HttpClient;
+import com.linecorp.armeria.common.AggregatedHttpMessage;
+import com.linecorp.armeria.common.HttpHeaderNames;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.testing.server.ServerRule;
+
+public class RedirectServiceTest {
+    private static int serverRule1Port;
+
+    static final HttpService SERVICE_BRANCH_1 = new AbstractHttpService() {
+        @Override
+        protected HttpResponse doGet(ServiceRequestContext ctx, HttpRequest req) throws Exception {
+            return HttpResponse.of(HttpStatus.OK, MediaType.PLAIN_TEXT_UTF_8, "SERVICE_BRANCH_1");
+        }
+    };
+    static final HttpService SERVICE_BRANCH_2 = new AbstractHttpService() {
+        @Override
+        protected HttpResponse doGet(ServiceRequestContext ctx, HttpRequest req) throws Exception {
+            return HttpResponse.of(HttpStatus.OK, MediaType.PLAIN_TEXT_UTF_8, "SERVICE_BRANCH_2");
+        }
+    };
+    static final HttpService SERVICE_NAMED_PARAM = new AbstractHttpService() {
+        @Override
+        protected HttpResponse doGet(ServiceRequestContext ctx, HttpRequest req) throws Exception {
+            String value1 = ctx.pathParam("var1");
+            String value2 = ctx.pathParam("var2");
+            String value3 = ctx.pathParam("var3");
+            return HttpResponse.of(HttpStatus.OK, MediaType.PLAIN_TEXT_UTF_8,
+                    "SERVICE_NAMED_PARAM " + value1 + " " + value2 + " " + value3);
+        }
+    };
+
+    @Rule
+    public ServerRule serverRule1 = new ServerRule(false) {
+        @Override
+        protected void configure(ServerBuilder sb) throws Exception {
+            try (ServerSocket ss = new ServerSocket(0)) {
+                serverRule1Port = ss.getLocalPort();
```

Comment:
FYI, you can declare this variable as local variable and then you can bring the port using `serverRule1.httpPort()` later outside of this scope.